app/src/main/java/helium314/keyboard/latin/spellcheck/AndroidWordLevelSpellCheckerSession.java

```diff
@@ -126,12 +132,40 @@ private void updateLocale() {
             mLocale = (null == localeString) ? null
                     : LocaleUtils.constructLocale(localeString);
             if (mLocale == null) mScript = ScriptUtils.SCRIPT_UNKNOWN;
-                else mScript = ScriptUtils.script(mLocale);
+            else mScript = ScriptUtils.script(mLocale);
+        }
+        if (!localesToCheck.contains(mLocale)) {
+            final List<Locale> updatedLocalesToCheck = new ArrayList<>();
+            updatedLocalesToCheck.add(mLocale);
+            // Make sure no other locales with mScript are added
+            for (int i = 0; i < localesToCheck.size(); i++) {
+                final Locale locale = localesToCheck.get(i);
+                if (!ScriptUtils.script(locale).equals(mScript)) {
+                    updatedLocalesToCheck.add(localesToCheck.get(i));
+                }
+            }
+            localesToCheck = updatedLocalesToCheck;
         }
     }
 
+    // Finds the enabled locale with a script matching one of the letters in the given text.
+    public Locale findLikelyLocaleOfText(final String text) {
```

Comment:
The name is somewhat misleading, better rename to reflect what it actualy does: find a locale with matching script

Reply:
Right.. maybe it should be moved out of the AndroidWordLevelSpellCheckerSession class too because it is not specifically related to the spellchecker except for the local variable localesToCheck (which can be passed as an argument)

Reply:
I think it's not necessary to move it, but I definitely agree with your reasoning. Possibly could be moved to `ScriptUtils` or maybe `LocaleUtils`.